### PR TITLE
Remove Java based frontend

### LINKED ISSUE #2119: Migrate S1314 (no-octal) to eslint implementation (new rule for TS)

### LINKED ISSUE #2393: Migrate S2260 (parsing error check)
Technically this class exists only to register the rule, as parsing error mechanism for eslint rules is handled differently. To "migrate" this rule, we just need to change `extends` clause to implement directly `JavaScriptCheck`, which will be true marker interface (without any methods)

### PR DESCRIPTION
**Pull request by saberduck**
Fix #2119
Fix #2393

### CODE REVIEW

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/NoSonarTest.java [hunk 1]
```diff
@@ -54,6 +56,12 @@ public static void startServer() {
     orchestrator.executeBuild(build);
   }
 
+  static WsClient newWsClient(Orchestrator orchestrator) {
```
Comment (guillaume-dequenne-sonarsource) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/NoSonarTest.java:59]: _Nitpick:_ Can we not use the one defined in `Tests` (`com.sonar.javascript.it.plugin.Tests.newWsClient`) instead ?
I see this one is used in `EslintCustomRulesTest` as well, providing a different instance of orchestrator, but perhaps we can still only keep one method?

File: sonar-javascript-plugin/pom.xml [hunk 2]
```diff
@@ -73,6 +61,10 @@
       <groupId>org.sonarsource.sonarqube</groupId>
       <artifactId>sonar-testing-harness</artifactId>
     </dependency>
+    <dependency>
```
Comment (andrea-guarino-sonarsource) [inline sonar-javascript-plugin/pom.xml:64]: why do we need this dependency?

**Review by andrea-guarino-sonarsource — APPROVED** [review]
LGTM, as already discussed, there are still some js test files that could be removed.

File: sonar-javascript-plugin/pom.xml (hunk 2 above)
Comment (guillaume-dequenne-sonarsource) [inline sonar-javascript-plugin/pom.xml:64]: I see it's already in the parent pom but I don't understand why it's needed here?

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
LGTM, left minor comments

File: sonar-javascript-plugin/pom.xml (hunk 2 above)
Comment (saberduck) [reply sonar-javascript-plugin/pom.xml]: indeed, it seems it can be removed

**Comment by saberduck** [comment]
This change was reverted in #2417 because custom rules were deprecated in current LTS cycle of SonarQube. Java based frontend will be removed in SQ 9.x cycle

**Pull request merged**